### ISSUE TITLE
Too much jank on `Stripe.instance.createPaymentMethod` halts further execution.

### ISSUE THREAD

**Issue by asadamatic** [flutter-stripe/flutter_stripe#1244]
**Describe the bug**
I'm using [flutter_stripe][1] package and I have functionality in my app to save a card, so it can be used later. I'm following these steps;

 1. Calling Stripe.instance.createPaymentMethod
 2. Calling Stripe.instance.createToken
 3. Posting the Token received in step 2 to RestAPI.

> Logs suggest that my app takes a lot of time to process steps 1 and
> causes jank and once that is done, the app halts the execution of the
> rest of the steps.

**To Reproduce**

**Step 1**

```

await Stripe.instance.createPaymentMethod(
        params: PaymentMethodParams.card(
          paymentMethodData: PaymentMethodData(
            billingDetails:
                BillingDetails(name: cardHolderName), // it didn't work
          ),
        ),
        options: const PaymentMethodOptions(
            setupFutureUsage: PaymentIntentsFutureUsage.OnSession));
```

Step 2, 3

```
  await Stripe.instance
        .createToken(const CreateTokenParams.card(
            params: CardTokenParams(
      type: TokenType.Card,
    
         ))).then((value) async {
    
      print('Token Received');

      // Calling API
      await createCard(value.id);
    
    }).onError((error, stackTrace) {
     getErrorDialog(error);
    });
  }
```

This problem is occurring only on Android.

I tried using `Isolates` but it didn't affect the performance, probably because I couldn't configure it well. So any guide on using Isolates for my case would be helpful (Please don't share medium articles, as I've already seen some).

**Logs** (During the execution of this function)
```
I/flutter (18568): Started Execution
I/ViewRootImpl(18568): jank_removeInvalidNode jank list is null
I/ViewRootImpl(18568): jank_removeInvalidNode jank list is null
I/ViewRootImpl(18568): jank_removeInvalidNode jank list is null
I/ViewRootImpl(18568): jank_removeInvalidNode jank list is null
I/ViewRootImpl(18568): jank_removeInvalidNode jank list is null
I/ViewRootImpl(18568): jank_removeInvalidNode jank list is null
I/ViewRootImpl(18568): jank_removeInvalidNode jank list is null
I/flutter (18568): Pay Method Created
I/Choreographer(18568): Skipped 73 frames!  The application may be doing too much work on its main thread.
I/ViewRootImpl(18568): jank_removeInvalidNode jank list is null
I/ViewRootImpl(18568): jank_removeInvalidNode jank list is null
E/flutter ( 1627): [ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: type 'int' is not a subtype of type 'String' in type cast
E/flutter ( 1627): #0      _$$_TokenDataFromJson (package:stripe_platform_interface/src/models/create_token_data.g.dart:146:40)
E/flutter ( 1627): #1      new _$_TokenData.fromJson (package:stripe_platform_interface/src/models/create_token_data.freezed.dart:1777:7)
E/flutter ( 1627): #2      _$TokenDataFromJson (package:stripe_platform_interface/src/models/create_token_data.freezed.dart:1577:21)
E/flutter ( 1627): #3      new TokenData.fromJson (package:stripe_platform_interface/src/models/create_token_data.dart:144:7)
E/flutter ( 1627): #4      MethodChannelStripe.createToken.<anonymous closure> (package:stripe_platform_interface/src/method_channel_stripe.dart:287:44)
E/flutter ( 1627): #5      ResultParser.parse (package:stripe_platform_interface/src/result_parser.dart:14:24)
E/flutter ( 1627): #6      MethodChannelStripe.createToken (package:stripe_platform_interface/src/method_channel_stripe.dart:288:10)
E/flutter ( 1627): <asynchronous suspension>
E/flutter ( 1627): #7      Stripe.createToken (package:flutter_stripe/src/stripe.dart:286:25)
E/flutter ( 1627): <asynchronous suspension>
E/flutter ( 1627): #8      FutureExtensions.onError.<anonymous closure> (dart:async/future.dart:1049:15)
E/flutter ( 1627): <asynchronous suspension>
E/flutter ( 1627): 
```
**At this point, the execution of rest of the function is halted.**

**Smartphone / tablet** (Real Device)
 - Device: Honor 8C
 - OS: Android 8.1
 
 **Flutter Version**
Flutter 3.7.12 • channel stable • 
Engine • revision 1a65d409c7
Tools • Dart 2.19.6 • DevTools 2.20.1

**Additional context**
This performance issue does not occur on IOS simulator.

  [1]: https://pub.dev/packages/flutter_stripe

**Comment by asadamatic** [comment]
Tried on another device and didn't experience the jank inside `Stripe.instance.createPaymentMethod` , but got it error in `Stripe.instance.createToken` and then observed jank after that.

**Smartphone / tablet** (Real Device)
 - Device: Xioami A3
 - OS: Android 11.0
 

 ```
 E/flutter ( 1579): [ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: type 'int' is not a subtype of type 'String' in type cast
E/flutter ( 1579): #0      _$$_TokenDataFromJson (package:stripe_platform_interface/src/models/create_token_data.g.dart:146:40)
E/flutter ( 1579): #1      new _$_TokenData.fromJson (package:stripe_platform_interface/src/models/create_token_data.freezed.dart:1777:7)
E/flutter ( 1579): #2      _$TokenDataFromJson (package:stripe_platform_interface/src/models/create_token_data.freezed.dart:1577:21)
E/flutter ( 1579): #3      new TokenData.fromJson (package:stripe_platform_interface/src/models/create_token_data.dart:144:7)
E/flutter ( 1579): #4      MethodChannelStripe.createToken.<anonymous closure> (package:stripe_platform_interface/src/method_channel_stripe.dart:287:44)
E/flutter ( 1579): #5      ResultParser.parse (package:stripe_platform_interface/src/result_parser.dart:14:24)
E/flutter ( 1579): #6      MethodChannelStripe.createToken (package:stripe_platform_interface/src/method_channel_stripe.dart:288:10)
E/flutter ( 1579): <asynchronous suspension>
E/flutter ( 1579): #7      Stripe.createToken (package:flutter_stripe/src/stripe.dart:286:25)
E/flutter ( 1579): <asynchronous suspension>
E/flutter ( 1579): #8      FutureExtensions.onError.<anonymous closure> (dart:async/future.dart:1049:15)
E/flutter ( 1579): <asynchronous suspension>
E/flutter ( 1579): 
I/Choreographer( 1579): Skipped 33 frames!  The application may be doing too much work on its main thread.
I/Choreographer( 1579): Skipped 35 frames!  The application may be doing too much work on its main thread.
I/Choreographer( 1579): Skipped 38 frames!  The application may be doing too much work on its main thread.
I/Choreographer( 1579): Skipped 30 frames!  The application may be doing too much work on its main thread.
I/Choreographer( 1579): Skipped 31 frames!  The application may be doing too much work on its main thread.
I/Choreographer( 1579): Skipped 35 frames!  The application may be doing too much work on its main thread.
I/Choreographer( 1579): Skipped 38 frames!  The application may be doing too much work on its main thread.
I/Choreographer( 1579): Skipped 35 frames!  The application may be doing too much work on its main thread.
I/Choreographer( 1579): Skipped 30 frames!  The application may be doing too much work on its main thread.
I/Choreographer( 1579): Skipped 33 frames!  The application may be doing too much work on its main thread.
I/Choreographer( 1579): Skipped 32 frames!  The application may be doing too much work on its main thread.
I/Choreographer( 1579): Skipped 38 frames!  The application may be doing too much work on its main thread.
I/Choreographer( 1579): Skipped 37 frames!  The application may be doing too much work on its main thread.
.....
......
 ```

**Comment by remonh87** [comment]
Thanks for reporting. 
We are using Flutter platform view and unfortunately we have little control over the speed of the views since we only render native Stripe views in a platform view. I have to close this unfortunately.

**Comment by asadamatic** [comment]
@remonh87 Okay, but what about this error that pops up on `createToken` 

```
E/flutter ( 1579): [ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: type 'int' is not a subtype of type 'String' in type cast
E/flutter ( 1579): #0      _$$_TokenDataFromJson (package:stripe_platform_interface/src/models/create_token_data.g.dart:146:40)
E/flutter ( 1579): #1      new _$_TokenData.fromJson (package:stripe_platform_interface/src/models/create_token_data.freezed.dart:1777:7)
E/flutter ( 1579): #2      _$TokenDataFromJson (package:stripe_platform_interface/src/models/create_token_data.freezed.dart:1577:21)
E/flutter ( 1579): #3      new TokenData.fromJson (package:stripe_platform_interface/src/models/create_token_data.dart:144:7)
E/flutter ( 1579): #4      MethodChannelStripe.createToken.<anonymous closure> (package:stripe_platform_interface/src/method_channel_stripe.dart:287:44)
E/flutter ( 1579): #5      ResultParser.parse (package:stripe_platform_interface/src/result_parser.dart:14:24)
E/flutter ( 1579): #6      MethodChannelStripe.createToken (package:stripe_platform_interface/src/method_channel_stripe.dart:288:10)
E/flutter ( 1579): <asynchronous suspension>
E/flutter ( 1579): #7      Stripe.createToken (package:flutter_stripe/src/stripe.dart:286:25)
E/flutter ( 1579): <asynchronous suspension>
E/flutter ( 1579): #8      FutureExtensions.onError.<anonymous closure> (dart:async/future.dart:1049:15)
E/flutter ( 1579): <asynchronous suspension>
E/flutter ( 1579): 
```

**Comment by remonh87** [comment]
That is a duplicate of #1112
